### PR TITLE
Added more types for functions called from unit tests

### PR DESCRIPTION
**Pull request by tisnik**
## Type of change

- [x] Refactor
- [ ] New feature
- [ ] Bug fix
- [ ] CVE fix
- [ ] Optimization
- [ ] Documentation Update
- [ ] Configuration Update
- [ ] Bump-up dependent library

## Description

Added more types for functions called from unit tests (and thus tested properly)

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] Unit tests passed locally.
- [ ] If it is a core feature, I have added thorough tests.

## Testing
- To be checked on CI

### CODE REVIEW

File: src/query_helpers/question_validator.py [hunk 1]
```diff
@@ -18,7 +18,7 @@ def __init__(self):
         """
         self.logger = Logger("question_validator").logger
 
-    def validate_question(self, conversation, query, **kwargs):
+    def validate_question(self, conversation: str, query: str, **kwargs) -> list[str]:
```
Comment (thoraxe) [inline src/query_helpers/question_validator.py:21]: Does this return a list of strs or just a single str?

Comment (bparees) [reply src/query_helpers/question_validator.py:21]: list/array: https://github.com/openshift/lightspeed-service/pull/81/files#diff-9ca3ebc421243362c6b59edda5939229ef8cd3c3cd4c83753986de3b315d3092L66

**Pull request merged**